### ISSUE TITLE
question: getting disk iops metrics

### ISSUE THREAD

**Issue by LazarenkoA** [shirou/gopsutil#946]
Please tell me how can I get disk iops records using your project? I'm talking about a similar performance monitor (windows) `Disk Transfers / sec`

I try
```go
d, _ := disk.IOCounters()
for k, v := range d {
    fmt.Printf("disk %s, data: %v\n", k, v)
}
```

output
```
disk C:, data: {"readCount":15317568,"mergedReadCount":0,"writeCount":29402968,"mergedWriteCount":0,"readBytes":853855299072,"writeBytes":1328147898368,"readTime":63485,"writeTime":47262,"iopsInProgress":0,"ioTime":0,"weightedIO":0,"name":"C:","serialNumber":"","label":""}
```

iopsInProgress - always 0

**Comment by LazarenkoA** [comment]
the question is relevant for linux too

**Comment by Lomanic** [comment]
`IOCountersStat.IopsInProgress` is only set on linux (for the moment at least):
```
rg 'IopsInProgress'
disk/disk_linux.go
432:                    IopsInProgress:   iopsInProgress,

disk/disk.go
40:     IopsInProgress   uint64 `json:"iopsInProgress"`
```

Implementing the same thing on windows would involve pdh/performance counters, that are hard to use.

> the question is relevant for linux too

Do you mean you have the same issue (value set to 0) on linux or you want an equivalent of `Disk Transfers/sec` there?

`Disk Transfers/sec` is defined as the following [in this MS document](https://www.oreilly.com/library/view/windows-server-2008/30000LTI00084/30000LTI00084_ch15sec22.html)

> Physical Disk\Disk Transfers/sec
>
> The Disk Transfers/sec is the number I/O request Packets (IRP) that has been completed during the sampling interval. A disk can only handle one I/O operation at a time. So the number of physical disks attached to the computer serves as an upper bound on the sustainable disk IO rate. Where disk arrays are used, divide the Disk Transfers/sec by the number of disks in the array to estimate individual disk I/O rates.

I don't think linux provides these stats [in /sys/block/<dev>/stat](https://www.kernel.org/doc/Documentation/block/stat.txt) (not used in gopsutil), the most approaching would be one of the following fields
```
		12 - I/Os currently in progress
		13 - time spent doing I/Os (ms)
		14 - weighted time spent doing I/Os (ms)
```

which are extracted as `IOCountersStat.IopsInProgress`, `IOCountersStat.IoTime` and `IOCountersStat.WeightedIO` respectively.

But once again, it's hard to reason about disks the same way in linux and windows, as concepts are really different between the two OSes.

**Comment by LazarenkoA** [comment]
>Do you mean you have the same issue (value set to 0) on linux or you want an equivalent of Disk Transfers/sec there?
yes

**Comment by Lomanic** [comment]
This reply doesn't answer the question :/

**Comment by LazarenkoA** [comment]
> This reply doesn't answer the question :/

yes on linux iopsInProgress i get always 0, yes i would like to get iops on linux.
of the proposed
```
12 - I/Os currently in progress
13 - time spent doing I/Os (ms)
14 - weighted time spent doing I/Os (ms)
```
13, 14 this is a time metric, I don't understand how it can suit me
